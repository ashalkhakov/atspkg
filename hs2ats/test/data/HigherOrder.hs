data GenericSum a b = GenericL a
                    | GenericR b

newtype GenericSumInt = GenericSumInt (GenericSum String Int)
