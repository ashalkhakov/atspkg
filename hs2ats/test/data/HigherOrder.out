datavtype generic_sum(a: vt@ype, b: vt@ype) =
  | Generic_l of a
  | Generic_r of b

vtypedef generic_sum_int = generic_sum(Strptr1, int)
